gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -11,186 +11,232 @@ You may obtain a copy of the License at
 https://github.com/GPII/universal/blob/master/LICENSE.txt
 */
 
-"use strict";
-
-var fluid = require("infusion");
-var crypto = require("crypto");
-
-var gpii = fluid.registerNamespace("gpii");
-fluid.registerNamespace("gpii.oauth2");
-
-fluid.defaults("gpii.oauth2.authorizationService", {
-    gradeNames: ["fluid.component"],
-    urls: {
-        preferencesPost: ""
-    },
-    components: {
-        dataStore: {
-            type: "gpii.oauth2.dataStore"
+var fluid = fluid || require("infusion");
+
+(function () {
+
+    "use strict";
+
+    var gpii = fluid.registerNamespace("gpii");
+
+    fluid.defaults("gpii.oauth2.authorizationService", {
+        gradeNames: ["fluid.component"],
+        urls: {
+            preferencesPost: ""
         },
-        tempPreferencesDataSource: { // TODO: This should be removed - see GPII-1521
-            type: "kettle.dataSource.URL",
-            options: {
-                url: "{authorizationService}.options.urls.preferencesPost",
-                termMap: {
-                    view: "%view"
-                },
-                writable: true,
-                writeMethod: "POST"
+        components: {
+            dataStore: {
+                type: "gpii.oauth2.dataStore"
+            },
+            codeGenerator: {
+                type: "gpii.oauth2.codeGenerator"
+            },
+            tempPreferencesDataSource: { // TODO: This should be removed - see GPII-1521
```

Comment:
Will address this in a separate pull request using the JIRA ticket https://issues.gpii.net/browse/GPII-1521